gui-tk/wiring.tcl

```diff
@@ -809,6 +810,46 @@ proc contextMenu {x y X Y} {
     tk_popup .wiring.context $X $Y
 }
 
+proc setupPickDimMenu {} {
+    global dimLabelPicked
+    if {![winfo exists .wiring.context.pick]} {
+        toplevel .wiring.context.pick
+        wm title .wiring.context.pick "Pick two dimensions"
+        frame .wiring.context.pick.select
+        scrollbar .wiring.context.pick.select.vscroll -orient vertical -command {
+            .wiring.context.pick.select.lb yview}
+        listbox .wiring.context.pick.select.lb -listvariable dimLabelPicked \
+            -selectmode extended -selectforeground blue \
+            -width 35 \
+            -yscrollcommand {.wiring.context.pick.select.vscroll set} 
+        pack .wiring.context.pick.select.lb -fill both  -expand y -side left
+        pack .wiring.context.pick.select.vscroll -fill y -expand y -side left
+        pack .wiring.context.pick.select
+        buttonBar .wiring.context.pick {
+            set pick {}
+            foreach i [.wiring.context.pick.select.lb curselection] {
```

Comment:
curelection returns a list, so no need to iteratively build a new list, just assign it with a set statement

Reply:
I don't why, but when I assign [.wiring.context.pick.select.lb curselection] directly to "dimLabelPick", and then append it to "pick", only the indices I picked appear, which obviously don't agree with the labels themselves, so the first two labels are always chosen. I've tried to do this in several different way but the result is also the same... For some reason dimLabelPicked is not assigned correctly...

Reply:
My apologies - I thought curselection returned the selected values. I misread the documentation - I see now it returns indices, so your proposed code is as good as any.